mythril/ethereum/util.py

```diff
@@ -168,18 +168,72 @@ def parse_pragma(solidity_code):
     all_versions = solcx.get_installed_solc_versions()
 
 
+# remove_comments_strings(str):
+# Return str without Solidity comments and strings
+import re
+
+VOID_START = re.compile("//|/\\*|\"|'")
+QUOTE_END = re.compile("(?<!\\\\)'")
+DQUOTE_END = re.compile('(?<!\\\\)"')
+
+
+def remove_comments_strings(todo):
+    done = ""
+    while True:
+        m = VOID_START.search(todo)
+        if not m:
+            done += todo
+            break
+        else:
+            done += todo[: m.start()]
+            if m[0] == "//":
+                end = todo.find("\n", m.end())
+                todo = "" if end == -1 else todo[end:]
+            elif m[0] == "/*":
+                end = todo.find("*/", m.end())
+                done += " "
+                todo = "" if end == -1 else todo[end + 2 :]
+            else:
+                if m[0] == "'":
+                    m2 = QUOTE_END.search(todo[m.end() :])
+                else:
+                    m2 = DQUOTE_END.search(todo[m.end() :])
+                if not m2:
+                    # unclosed string
+                    break
+                todo = todo[m.end() + m2.end() :]
+    return done
+
+
 def extract_version(file: typing.Optional[str]):
     if file is None:
         return None
     version_line = None
-    for line in file.split("\n"):
-        if "pragma solidity" not in line:
-            continue
-        version_line = line.rstrip()
-        break
+
+    # normalize line endings
```

Comment:
This part of extracting the version_line can now be moved to separate function. It'll be easier to read and understand.